spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindReturnRef.java

```diff
@@ -104,15 +118,26 @@ public void visit(Method obj) {
     }
 
     @Override
-    public void visit(Code obj) {
-        if (check) {
-            super.visit(obj);
+    public void sawOpcode(int seen) {
+        if (isFirstVisit) {
```

Comment:
Are you sure this still detects true positives when the second assignment is modifiable?
I probably misunderstood but it looks like only the first value is considered, so it might miss true positives when only the first assignment was indeed unmodifiable. 
In any case, it would be beneficial to add tests for that case.

Reply:
Thanks, I added some tests with if-elses. It should have found these, if the condition in line 217-218 would have been okay. I fixed, so now it works for the current tests, however when using the ternary operator, it does not work properly, I'm still looking into that issue.

Reply:
The logic here is that the examined class is visited two times: the first time, every value assignment is collected. Then the bugs are reported with consideration of the collected data.

Reply:
Thanks for the explanation and the unit tests showing the true positives!